From e5c41452f2acaf658072150f53f645ffc0ebfccc Mon Sep 17 00:00:00 2001
From: Paul Cercueil <paul@crapouillou.net>
Date: Thu, 30 Sep 2021 09:24:08 +0100
Subject: [PATCH 15/19] serial: Implement {get,set}_trigger callback

These two were missing, which meant it was impossible to configure a
device's trigger.

Signed-off-by: Paul Cercueil <paul@crapouillou.net>
---
 serial.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/serial.c b/serial.c
index 99966c3c..6fe4a11e 100644
--- a/serial.c
+++ b/serial.c
@@ -355,6 +355,24 @@ static int serial_set_timeout(struct iio_context *ctx, unsigned int timeout)
 	return 0;
 }
 
+static int serial_get_trigger(const struct iio_device *dev,
+			      const struct iio_device **trigger)
+{
+	const struct iio_context *ctx = iio_device_get_context(dev);
+	struct iio_context_pdata *pdata = iio_context_get_pdata(ctx);
+
+	return iiod_client_get_trigger(pdata->iiod_client, NULL, dev, trigger);
+}
+
+static int serial_set_trigger(const struct iio_device *dev,
+			      const struct iio_device *trigger)
+{
+	const struct iio_context *ctx = iio_device_get_context(dev);
+	struct iio_context_pdata *pdata = iio_context_get_pdata(ctx);
+
+	return iiod_client_set_trigger(pdata->iiod_client, NULL, dev, trigger);
+}
+
 static const struct iio_backend_ops serial_ops = {
 	.get_version = serial_get_version,
 	.open = serial_open,
@@ -369,6 +387,8 @@ static const struct iio_backend_ops serial_ops = {
 	.shutdown = serial_shutdown,
 	.get_description = serial_get_description,
 	.set_timeout = serial_set_timeout,
+	.get_trigger = serial_get_trigger,
+	.set_trigger = serial_set_trigger,
 };
 
 static const struct iiod_client_ops serial_iiod_client_ops = {
-- 
2.30.2

